cdap-common/src/main/java/co/cask/cdap/common/namespace/Namespaces.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common.namespace;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.proto.Id;
+import org.apache.twill.filesystem.Location;
+import org.apache.twill.filesystem.LocationFactory;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Utilities for namespaces
+ */
+public class Namespaces {
+  private Namespaces() {
+  }
+
+  public static Location getNamespaceHomeLocation(LocationFactory locationFactory, CConfiguration cConf,
```

Comment:
Ok, I tried it. 
The callers of the method in Programs class would need updating to pass in a different locationFactory.
and I think its better to just modify Programs class, instead of the 4 places that use that method.